hydra-cluster/src/CardanoCluster.hs

```diff
@@ -75,35 +50,12 @@ data RunningCluster = RunningCluster ClusterConfig [RunningNode]
 defaultNetworkId :: NetworkId
 defaultNetworkId = Testnet (NetworkMagic 42)
 
--- FIXME: This is hard-coded and should correspond to the initial funds set in
--- the genesis file.
+-- NOTE: This is hard-coded and needs to correspond to the initial funds set in
+-- the genesis-shelley.json file.
 availableInitialFunds :: Num a => a
 availableInitialFunds = 900_000_000_000
 
--- | Configuration parameters for the cluster.
-data ClusterConfig = ClusterConfig
-  { parentStateDirectory :: FilePath
-  , networkId :: NetworkId
-  , initialFunds :: [VerificationKey PaymentKey]
-  }
-
-asSigningKey :: AsType (SigningKey PaymentKey)
-asSigningKey = AsSigningKey AsPaymentKey
-
-withCluster ::
-  Tracer IO ClusterLog -> ClusterConfig -> (RunningCluster -> IO ()) -> IO ()
-withCluster tr cfg@ClusterConfig{parentStateDirectory, initialFunds} action = do
-  systemStart <- initSystemStart
-  (cfgA, cfgB, cfgC) <-
-    makeNodesConfig parentStateDirectory systemStart
-      <$> randomUnusedTCPPorts 3
-
-  withBFTNode tr cfgA initialFunds $ \nodeA -> do
-    withBFTNode tr cfgB initialFunds $ \nodeB -> do
-      withBFTNode tr cfgC initialFunds $ \nodeC -> do
-        let nodes = [nodeA, nodeB, nodeC]
-        action (RunningCluster cfg nodes)
-
+-- | Enumeration of known actors for which we can get the 'keysFor' and 'writeKeysFor'.
 data Actor
```

Comment:
question: Do we really need the actors to be limited here, now that we don't need to know their keys beforehand? IIRC, we did that when the keys were used to manually populate `initialFunds` section in the genesis file but it's not relevant anymore.

Reply:
They still enumerate the keys available in `hydra-cluster/config/credentials`. As long as we have some keys hard-coded there, I would like to keep this enumeration.

Reply:
But that's the point I make: We can remove those keys now, can't we?